prow/repoowners/endpoint.go

```diff
@@ -0,0 +1,131 @@
+package repoowners
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+type OwnersServer struct {
+	OwnersClient *Client
+}
+
+type ownersRequest struct {
+	Org        string   `json:"org_name"`
+	Repo       string   `json:"repo_name"`
+	BaseCommit string   `json:"base_commit"`
+	Paths      []string `json:"file_paths"`
+}
+
+type httpJSONError struct {
+	Message string `json:"error_message"`
+}
+
+type ownersResponse struct {
+	ConfigMap map[string]*Config `json:"owners,omitempty"`
+}
+
+const (
+	JsonContentType string = "This JSON-API required a JSON content-type"
+	EmptyBody       string = "The body is empty, there is no content to parse"
+	InternalError   string = "The server entered in a failed state"
+	EmtpyPath       string = "Path(s) list is empty or non-existing"
+	NonPossibleRepo string = "Please provide corret git information for Org/Repo and Base commit"
+	CorruptedBody   string = "The body seem to be corrupted"
+	CorruptedJSON   string = "The JSON seem to be corrupted"
+)
+
+func generateMessage(message string) []byte {
+	result, err := json.Marshal(httpJSONError{Message: message})
+	if err != nil {
+		result = []byte(fmt.Sprintf("FATAL : %v", err))
+	}
+
+	return result
+}
+
+func (s *OwnersServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	if r.Header.Get("Content-Type") != "application/json" {
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write(generateMessage(JsonContentType))
+		return
+	}
+
+	rawBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
```

Comment:
Yes, we already (almost) speak about that [here](https://github.com/improbable/k8s-test-infra/pull/10#discussion_r238694866), the point is that, if we do so, we will show directly some information regarding our FS (like you mentioned before) but [the suggestion](https://github.com/improbable/k8s-test-infra/pull/10#discussion_r238735411) from Peter go into the same direction as you. 

I am not really comfortable to print any FS information before actually having a proper auth except if our cluster is secured enough to handle it for the moment ?

Reply:
@petemounce and I just had a chat about this, whereupon we realized that:

1. This is a new endpoint on an existing service.
1. That existing service is NOT directly available outside of the `corp-eu3` cluster
1. All exposed endpoints on `hook` now are exposed by a proxy.
1. The only client will be residing in the same cluster as the service.

Since the only client that needs to access this new endpoint is cluster-local, we just won't add a proxy configuration for the new endpoint.  Cascader will be able to access the new endpoint directly with no issues.

Thus, adding this as a new endpoint, and not modifying the existing endpoints, means it's safe to just return the data and not worry about auth or any information disclosure (I'd feel differently if this service could perform potentially destructive actions, but it's read-only).